sessions.md

```diff
@@ -0,0 +1,210 @@
+
+* [Options](#Options)
+  * [httpOnly](#http-only)
+  * [secure](#secure)
+  * [path](#path)
+  * [domain](#domain)
+  * [validationStrategy](#validation-strategy)
+* [Incoming Request Handling](#incoming-request-handling)
+* [/me Handling](#me-handling)
+
+
+## Feature Description
+
+This document describes the how our framework integrations should handle user
+sessions and session management.
+
+There are a few things we'll do in respect to user sessions:
+
+- We'll always store an access and refresh token in the browser,
+these tokens will be obtained by using the password grant flow on
+the configured Stormpath application
+
+- This cookie will always be set with the `HttpOnly` flag set, meaning that no
+  Javascript access will be allowed to this cookie.
+
+- This cookie will be set with the `secure` flag if the application is in
+  production -- this ensures that no cookies will be set over plain old HTTP
+  (*leaking information to potential attackers*).
+
+- The cookies will always be named `accessToken` and `refreshToken`.
+
+- The Expiration time each cookie should be determined by the `exp` claim of
+the token, and this value is controlled by the Oauth Policy of the Application
+
+**NOTE**: We do NOT need to support server-side session storage, because the way
+this works is like so: a JWT is created which contains an account href.  The
+account href is persisted in our internal Stormpath SDK cache.  This means no
+server side session storage is necessary as the caching optimizations are
+already handled internally (*by us*).
+
+
+## <a name="Options"></a> Options
+
+This table is a list of all the options that are required by this feature.
+Detailed descriptions follow.  How the option names are translated into the
+framework language (e.g. to camel case, or not)? Is not specified here.
+
+```json
+{
+  "session": {
+    "httpOnly": true,
+    "secure": null,
+    "path": "/",
+    "domain": null,
+    "validationStrategy": "stormpath"
+  }
+}
+```
+
+#### <a name="http-only"></a> httpOnly
+
+This option determines whether or not this cookie can be read by Javascript code
+on the browser.
+
+**NOTE**: Disabling this option is a potential security issue, as your access
+tokens could be hijacked via XSS.
+
+<a href="#top">Back to Top</a>
+
+
+#### <a name="secure"></a> secure
+
+This option determines whether or not our cookies can be set over a non-https
+connection.  By default, this option is null, which means we will attempt to
+inspect the protocol of the incoming request, and make a decision of what to do
+automatically.
+
+Defining this value forces the `secure` mode to be either enabled or disabled
+all the time.
+
+**NOTE**: Disabling this option is a potential security issue, as setting
+cookies over a non-https connection is ripe for traffic inspection attacks.
+
+<a href="#top">Back to Top</a>
+
+
+#### <a name="path"></a> path
+
+This option determines what URI paths will be able to read our cookies.  By
```

Comment:
The default path should be the default 'context path' or 'base path' for the application deployed, not the domain root.

For example, let's say I have two apps deployed and accessible under the same domain:

https://myserver.com/appA
https://myserver.com/appB

For app A, its 'context path' (its 'base path' if you will) is `/appA`.  For app B it's `/appB`

Cookies should only be set for the relevant application by default and not readable by other apps on the same web infrastructure unless explicitly enabled.  This is relevant because HTTP reverse proxies might make two different apps available under the same domain, but those apps perhaps should never see each other's cookies - it is potentially a security risk to default to `/`.  If desired by the end-user, they should explicitly enable `/` as they are making a conscious decision to do that.

In Java at least, this is totally automated: if the app is deployed to the domain root, then the cookie path will be `/`.  If the app is deployed under `/appA`, the cookie path is `/appA`.